### DISCUSSION TITLE
Sharp corners

### DISCUSSION THREAD

**Discussion by K-pah** [OrcaSlicer/OrcaSlicer#6913] [Ideas]
I was thinking the other day about a printer's ability to draw sharp corners and the ability of input shaping to control ghosting. I then had a thought that may or may not be stupid, but here it is:

What if instead of following the regular path a sharp corner would, the tool head would follow this kind of path:
![image](https://github.com/user-attachments/assets/c6b5c2e3-71f7-4ce0-b647-4ba512fee271)
Black line represents the actual print, orange is where the toolhead would lift, green arrow is a forward motion and red would be backwards travel.

The drawing may not be the clearest representation so: toolhead lifts at the end of the line and overshoots and retracts to not leave any blobs. it follows the orange line to go back and align with the next path, pretravels along path mitigating any sideways vibrations, lowers toolhead at the start of next segment and continues printing.

I imagine this being a feature that can be toggled.

I reckon that while it would add time to the print, it may also mitigate ghosting and rounded corners as well. I am not aware if this idea has been brought up before so I apologize if it has.
I have no prior experience with coding and would not have the slightest idea where to start, however, if anybody thinks the idea is worth anything and has the know-how to implement it, maybe it can be given a shot.
Thank you.

**Comment by foreachthing** [comment]
> Toolhead lifts at the end of the line and overshoots and retracts to not leave any blobs

Stringing? 🧐
And, it would only be as sharp as the diameter of the nozzle anyway. Right?
Try it with a manually written gcode example and fine-tune. It would be interesting to see actual results.

**Reply by K-pah** [reply]
As I said, I do not currently have enough knowledge to do the test myself. I would if I did. I reckon well tuned PA and good retraction settings would reduce stringing as the toolhead would not stop at the end of the line like it generally would, but ramp away on the shown path.
Of course the corner's resolution is still dictated by the nozzle diameter, but any other influences that would round the corner or introduce ghosting would be minimised (maybe).
